test/integration-new/README.md

```diff
@@ -60,4 +60,237 @@ Currently the package is named as `integration-new` because we already have `int
+	input := ...
+
+	By("getting the output for xyz using input ")
+	output, err := func_xyz(input)
+	Expect(err).ToNot(HaveOccurred())
+ 
+        ...
+  
+
+})
+
+//The following function has checks and setup needed after running the suite.
+var _ = AfterSuite(func() {
+
+        ...
+
+})
+
+```
+### Structure of tests corresponding to each suite
+
+
+#### Logic Components
+
+- ```It```: Individual spec specified by It. It is the innermost component that holds the core testing logic. The other components listed below are hierarchically arranged before and after It in order to provision/deprovision the setup required to run the individual spec (It).
+- ```Describe``` : This blocks is used describe the individual behaviors of code.
+- ```Context``` : Context block is used to execute the behavior used by Describe block under different scenarios 
+- ```JustBeforeEach``` : Executed immediately before each test, however following the execution order from outside blocks to inside blocks before an It(spec) in case of multipe JustBeforeEach blocks.
+- ```JustAfterEach``` : Executed immediately after each test, however following the execution order from inside blocks to outside blocks after an It(spec) in case of multipe JustAfterEach blocks.
+- ```BeforeEach``` : Executed (not immediately) before each test, however following the execution order from outside blocks to inside blocks before an It(spec) in case of multipe BeforeEach blocks.
+- ```AfterEach``` : Executed (not immediately) after each test, however following the execution order from inside blocks to outside blocks after an It(spec) in case of multipe AfterEach blocks.
+
+Each of the above components are arranged hierarchically in a way that makes most sense for abstracting the common logic from the rest of the code. 
+
+Every ```BeforeEach``` precedes every ```JustBeforeEach``` before execution of an It.
+Every ```JustAfterEach``` precedes every ```AfterEach``` after execution of an It.
+
+
+Below is a sample test structure and may largely vary based on requirement. Some of the functions used below are just an example for illustration of functionality. Additions or deletions to the below snippet may be required for any new test.
+
+
+```go
+package <package_name>
+
+import (
+        // fmt is imported for printing
+	"fmt"
+	"strconv"
+
+	// The below folders are similar to the ones discussed above
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	
+	//ginkgo and the assertion library: gomega are imported below
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	
+	///misc packages imported eg: Node libraries etc
+	v1 "k8s.io/api/apps/v1"
+        ...
+)
+
+// This blocks is used describe the individual behaviors of code eg: "test pod networking"
+var _ = Describe("test sample xyz", func() {
```

Comment:
I think it would be helpful to describe a real example from our existing cni tests. Dont have to describe every line but we can just pick one example from cni or ipamd test and explain how it invokes functions using the TestAgentImage.